src/driver.ts

```diff
@@ -87,9 +92,13 @@ export function generateFile(): types.Resolve {
     const localName = ref.replace(/^[^#]+/, '');
     const fileName = ref.replace(/#.+$/, '');
     const ymlFile = path.resolve(path.dirname(options.sourceFile), fileName);
-    const generatedFileName = `${path.basename(ymlFile).replace(/\.[^.]*$/, '')}.types.generated`;
-    const moduleName = makeModuleName(fileName);
+    const generatedFilePath = options.preserveRefPathStructure ? fileName : path.basename(ymlFile);
```

Comment:
could we name the thing to just `preservePathStructure` ? That to me sounds understandable